ros/src/control/scripts/thrust_control.py

```diff
@@ -1,35 +1,94 @@
 #! /usr/bin/python
 import rospy
-from shared_msgs.msg import can_msg, auto_control_msg, thrust_control_msg, thrust_status_msg, thrust_command_msg
+from shared_msgs.msg import auto_control_msg, thrust_control_msg, thrust_status_msg, thrust_command_msg
 from sensor_msgs.msg import Imu
 from std_msgs.msg import Float32
+import numpy as np
+import Complex_1
 
-def message_received(msg):
-  # This runs on a seperate thread from the pub
-  pass
-def command_received(comm):
-  # This runs on a seperate thread from the pub
-  pass
+desired_a = None
+desired_p = None
+locked_dims_list = None
+disabled_list = None
+inverted_list = None
+
+def _auto_command(msg):
+  global desired_a #desired thrust from automatic control
+  global locked_dims_list #locked dimensions
+  desired_a = msg.thrust_vec
+  locked_dims_list = msg.dims_locked
+
+def _pilot_command(comm):
+  global desired_p #desired thrust from pilot
+  global disabled_list #disabled thrusters
+  global inverted_list #inverted thrusters
+  desired_p = comm.desired_thrust
+  disabled_list = comm.disable_thrusters
+  inverted_list = comm.inverted
 
 if __name__ == "__main__":
+  
+  #initialize node and rate
   rospy.init_node('thrust_control')
+  rate = rospy.Rate(10) #10 hz
 
+  #initialize subscribers
   auto_sub = rospy.Subscriber('auto_control', 
-      auto_control_msg, message_received)
+      auto_control_msg, _auto_command)
   comm_sub = rospy.Subscriber('/surface/thrust_command',
-      thrust_command_msg, command_received)
-
+      thrust_command_msg, _pilot_command)
+  
+  #initialize publishers
   thrust_pub = rospy.Publisher('thrust_control',
-    thrust_control_msg, queue_size=10);
-
+    thrust_control_msg, queue_size=10)
   status_pub = rospy.Publisher('thrust_status',
-    thrust_status_msg, queue_size=10);
+    thrust_status_msg, queue_size=10)
+
+  #define variable for class Complex to allow calculation of thruster pwm values
+  c = Complex_1.Complex()
 
-  rate = rospy.Rate(10) # 10hz
-  # TODO: I2C related activities
   while not rospy.is_shutdown():
-    thrust_pub.publish(thrust_control_msg())
-    status_pub.publish(thrust_status_msg())
-    rate.sleep()
-    
+    if desired_a == None or desired_p == None:
+      desired_thrust_final = np.array([0.0, 0.0, 0.0, 0.0, 0.0, 0.0])
+      disabled_list = [False, False, False, False, False, False, False, False]
+      inverted_list = [0, 0, 0, 0, 0, 0, 0, 0]
+    elif desired_a == None and desired_p != None:
+      desired_thrust_final = desired_p
+    elif desired_a != None and desired_p == None:
+      desired_thrust_final = desired_a
+      disabled_list = [False, False, False, False, False, False, False, False]
+      inverted_list = [0, 0, 0, 0, 0, 0, 0, 0]
+    else:
```

Comment:
I think this would all be simpler if you just had an empty desired_thrust_final, and then did this 'else' section to fill it. Kinda makes the other if conditions unnecessary right? Correct me if I'm wrong.

Reply:
Sort of, that's similar to what I had before. I can at the very least use it to simplify the if conditions but I was trying to account for cases where the node hasn't received any instructions yet and therefore can't assign specific values to desired_thrust_final.

Reply:
Yeah, so if you make desired_thrust_final 8 zeroes (not empty, sorry), then if there are no commands to iterate through it will just push 0 power, which is what we want right?